### PR TITLE
kde: link frameworkintegration to kpackage

### LINKED ISSUE #76615: Can't install a new KDE theme
**Describe the bug**

Downloading and installing KDE look-and-feel themes fails silently.

**To Reproduce**
1. Open System Settings -> Look and Feel
2. Click "get new look and feel themes" 
3. Click "install" on any theme. Wait for installation to finish.
4. Notice that no new theme has actually been installed. 

**Expected behavior**

The theme would be installed.

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.3.18, NixOS, 20.03pre206632.b0bbacb5213 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(jon): `"home-manager, nixpkgs-20.03pre203529.e8f2764fe8d"`
 - channels(root): `"nixos-20.03pre206632.b0bbacb5213"`
 - nixpkgs: `/home/jon/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by pasqui23**
###### Motivation for this change
Fixes https://github.com/NixOS/nixpkgs/issues/76615 .
However, while it builds, I cannot have been able to find if it really fix the problem.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/kde-frameworks/frameworkintegration.nix [hunk 1]
```diff
@@ -14,4 +15,7 @@ mkDerivation {
     kwidgetsaddons libXcursor qtx11extras
   ];
   propagatedBuildInputs = [ kconfigwidgets kiconthemes ];
+  postInstall=''
+    ln -s {${kpackage},$out}/libexec
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kde-frameworks/frameworkintegration.nix:19]:
```suggestion
    ln -s ${kpackage/libexec $out/libexec
```

File: pkgs/development/libraries/kde-frameworks/frameworkintegration.nix [hunk 2]
```diff
@@ -2,7 +2,8 @@
   mkDerivation, lib,
   extra-cmake-modules,
   kbookmarks, kcompletion, kconfig, kconfigwidgets, ki18n, kiconthemes, kio,
-  knewstuff, knotifications, kpackage, kwidgetsaddons, libXcursor, qtx11extras
+  knewstuff, knotifications, kpackage, kwidgetsaddons, libXcursor, qtx11extras,
+  kpackage
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kde-frameworks/frameworkintegration.nix:6]:
```suggestion
  knewstuff, knotifications, kpackage, kwidgetsaddons, libXcursor, qtx11extras
```

duplicated.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by ttuegel** [comment]
It would be better to patch the source to fix the hardcoded paths. Qt applications can behave strangely when there are symbolic links on search paths. Do you know what files `frameworkintegration` is looking for at `$out/libexec`, or how it is searching?

**Comment by pasqui23** [comment]
https://github.com/NixOS/nixpkgs/issues/76615#issuecomment-775219587
provides more context

**Comment by ttuegel** [comment]
That doesn't really answer my question, and it leaves me more confused: In that issue, they put a link to `frameworkintegration` into `kpackage`, but here you are putting a link to `kpackage` into `frameworkintegration`. Which direction should the link go? What package needs what file?

**Comment by pasqui23** [comment]
Yeah, I've told you I haven't been able to test it.

@tteggel  Continue this discussion on that issue, I'm closing this pr

**Pull request closed without merging**